src/main/java/me/anxuiz/settings/bukkit/plugin/gui/SettingsMenuBuilder.java

```diff
@@ -0,0 +1,130 @@
+package me.anxuiz.settings.bukkit.plugin.gui;
+
+import com.google.common.collect.Sets;
+import me.anxuiz.settings.Setting;
+import me.anxuiz.settings.SettingManager;
+import me.anxuiz.settings.Toggleable;
+import me.anxuiz.settings.bukkit.PlayerSettings;
+import me.anxuiz.settings.bukkit.plugin.Commands;
+import me.anxuiz.settings.bukkit.plugin.Permissions;
+import me.anxuiz.settings.bukkit.plugin.SettingsCommand;
+import me.anxuiz.settings.bukkit.plugin.gui.clickables.Clickable;
+import org.bukkit.Bukkit;
+import org.bukkit.ChatColor;
+import org.bukkit.Material;
+import org.bukkit.Sound;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.inventory.InventoryClickEvent;
+import org.bukkit.event.inventory.InventoryCloseEvent;
+import org.bukkit.inventory.Inventory;
+import org.bukkit.inventory.ItemStack;
+
+import java.util.Set;
+
+public class SettingsMenuBuilder {
+
+    private static final Set<Setting> settings = Sets.newLinkedHashSet(PlayerSettings.getRegistry().getSettings());
+
+    private static final ItemStack BORDER_ICON = new ItemBuilder(Material.STAINED_GLASS_PANE).setDamage((short) 7).setName(ChatColor.GRAY + "#").createItem();
+    private static final boolean BORDER = true;
+
+    private static Inventory makeInventory(boolean border) {
+        int size = settings.size();
+        if (border) {
+            int slots = (int) ((Math.round((size / 7) + 0.5) + 2) * 9);
```

Comment:
The size is initially divided by 7 as the 2 slide slots are removed to make place for the border, so each row can now only hold 7 Settings. After rounding and adding the 2 extra slots, I need to multiply it by 9 to determine the amount of slots the inventory needs to build with